convert.go

```diff
@@ -4,112 +4,198 @@ import (
 	"fmt"
 	"net"
 	"strings"
+	"sync"
 
 	ma "github.com/jbenet/go-multiaddr"
 	utp "github.com/jbenet/go-multiaddr-net/utp"
 )
 
 var errIncorrectNetAddr = fmt.Errorf("incorrect network addr conversion")
 
+type AddrParser func(a net.Addr) (ma.Multiaddr, error)
+type MaddrParser func(ma ma.Multiaddr) (net.Addr, error)
+
+var maddrParsers map[string]MaddrParser
+var addrParsers map[string]AddrParser
+var addrParsersLock sync.Mutex
+
+func init() {
+	addrParsers = make(map[string]AddrParser)
+	maddrParsers = make(map[string]MaddrParser)
+
+	registerDefaultAddrParsers()
+	registerDefaultMaddrParsers()
+}
+
+func registerDefaultAddrParsers() {
+	funcs := map[string]AddrParser{
```

Comment:
What do you think about adding the default addr/maddr parsers using the public `RegisterAddressType`?

My thinking is that if it's awkward or undesirable to do so, then that may be a signal that the public API needs some massaging to be likeable enough to use for registering them internally too. Does that make sense?